src/main/java/service/DataConverterImpl.java

```
@@ -0,0 +1,23 @@
+package service;
+
+import core.basesyntax.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> lines) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (int i = 1; i < lines.size(); i++) {
+            String line = lines.get(i);
+            String[] split = line.split(",");
+            String operationCode = split[0].trim();
+            FruitTransaction.Operation op = FruitTransaction.Operation.fromCode(operationCode);
+            String fruitName = split[1].trim();
+            int amount = Integer.parseInt(split[2].trim());
```

The task requires handling invalid input data, including negative quantities. You should add a check here to ensure the `amount` is not negative before creating the `FruitTransaction` object. If it's negative, you should throw an exception to prevent processing invalid data.